tracer/contrib/elastictraced/elastictraced.go

```diff
@@ -0,0 +1,100 @@
+// Package elastictraced provides tracing for the Elastic Elasticsearch client.
+// Supports v3 (gopkg.in/olivere/elastic.v3), v5 (gopkg.in/olivere/elastic.v5)
+// but with v3 you must use `DoC` on all requests to capture the request context.
+package elastictraced
+
+import (
+	"bytes"
+	"errors"
+	"io/ioutil"
+	"net/http"
+	"strings"
+
+	"github.com/DataDog/dd-trace-go/tracer"
+	"github.com/DataDog/dd-trace-go/tracer/ext"
+)
+
+// TracedTransport is a traced HTTP transport that captures Elasticsearch spans.
+type TracedTransport struct {
+	service string
+	tracer  *tracer.Tracer
+	*http.Transport
+}
+
+// RoundTrip satisfies the RoundTripper interface, wraps the sub Transport and
+// captures a span of the Elasticsearch request.
+func (t *TracedTransport) RoundTrip(req *http.Request) (*http.Response, error) {
+	if pathIsTraced(req.URL.Path) {
+		span := t.tracer.NewChildSpanFromContext("elasticsearch.query", req.Context())
+		span.Service = t.service
+		span.Type = ext.AppTypeDB
+		defer span.Finish()
+		span.SetMeta("elasticsearch.method", req.Method)
+		span.SetMeta("elasticsearch.url", req.URL.Path)
+		span.SetMeta("elasticsearch.params", req.URL.Query().Encode())
+
+		if req.Body != nil {
+			buf, err := ioutil.ReadAll(req.Body)
+			if err != nil {
+				return nil, err
+			}
+			span.SetMeta("elasticsearch.body", string(buf))
+			req.Body = ioutil.NopCloser(bytes.NewBuffer(buf))
+		}
+
+		// Run the request using the standard transport.
+		res, err := t.Transport.RoundTrip(req)
+
+		if err != nil {
+			span.SetError(err)
+			return res, err
+		} else if res.StatusCode < 200 || res.StatusCode > 299 {
+			buf, err := ioutil.ReadAll(res.Body)
+			if err != nil {
+				// Status text is best we can do if if we can't read the body.
+				span.SetError(errors.New(http.StatusText(res.StatusCode)))
+			} else {
+				span.SetError(errors.New(string(buf)))
+			}
+			res.Body = ioutil.NopCloser(bytes.NewBuffer(buf))
+		}
+		span.SetMetric(ext.HTTPCode, float64(res.StatusCode))
```

Comment:
This might be a number, it is more legit to see it as a "code" ie. using SetMeta. "Metric" were more for graphable values  (but we aren't sure we will keep that in the future).